src/block/BaseCoral.php

```diff
@@ -65,6 +59,21 @@ public function isAffectedBySilkTouch() : bool{
 
 	public function isSolid() : bool{ return false; }
 
+	public function hasWater() : bool{
```

Comment:
This method is being exposed to the API, but I'm not sure its name is clear enough. I'm also doubtful it should be public, considering that it's intended for internal usage.

Reply:
Would "isFilledWithWater" be a better name?

Reply:
Not really, since it's looking around the block.